src/spikeinterface/generation/tests/test_hybrid_tools.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+
+from spikeinterface.core import Templates
+from spikeinterface.core.generate import (
+    generate_ground_truth_recording,
+    generate_sorting,
+    generate_templates,
+    generate_unit_locations,
+)
+from spikeinterface.preprocessing.motion import correct_motion, load_motion_info
+from spikeinterface.generation.hybrid_tools import (
+    estimate_templates_from_recording,
+    generate_hybrid_recording,
+)
+
+
+def test_generate_hybrid_no_motion():
+    rec, _ = generate_ground_truth_recording(sampling_frequency=20000, seed=0)
+    hybrid, _ = generate_hybrid_recording(rec, seed=0)
+    assert rec.get_num_channels() == hybrid.get_num_channels()
+    assert rec.get_num_frames() == hybrid.get_num_frames()
+    assert rec.get_num_segments() == hybrid.get_num_segments()
+    assert np.array_equal(rec.get_channel_locations(), hybrid.get_channel_locations())
+
+
+def test_generate_hybrid_with_sorting():
+    gt_sorting = generate_sorting(durations=[10], num_units=20, sampling_frequency=20000, seed=0)
+    rec, _ = generate_ground_truth_recording(durations=[10], sampling_frequency=20000, sorting=gt_sorting, seed=0)
+    hybrid, sorting_hybrid = generate_hybrid_recording(rec, sorting=gt_sorting)
+    assert rec.get_num_channels() == hybrid.get_num_channels()
+    assert rec.get_num_frames() == hybrid.get_num_frames()
+    assert rec.get_num_segments() == hybrid.get_num_segments()
+    assert np.array_equal(rec.get_channel_locations(), hybrid.get_channel_locations())
+    assert sorting_hybrid.get_num_units() == len(hybrid.templates)
+
+
+def test_generate_hybrid_motion(create_cache_folder):
+    cache_folder = create_cache_folder
+    rec, _ = generate_ground_truth_recording(sampling_frequency=20000, durations=[10], seed=0)
+    correct_motion(rec, folder=cache_folder / "motion")
+    motion_info = load_motion_info(cache_folder / "motion")
```

Comment:
You can us coorect_motion(output_motion_info=True) to avoid a disk write here.
No ?

Reply:
yes